core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -168,6 +168,23 @@ case class ExecutorLostFailure(execId: String) extends TaskFailedReason {
 
 /**
  * :: DeveloperApi ::
+ * The task failed because the executor that it was running on was prematurely terminated. The
+ * executor is forcibly exited but the exit should be considered as part of normal cluster
+ * behavior.
+ */
+@DeveloperApi
+case class ExecutorForTaskExited(
+    taskId: Long,
+    execId: String,
+    exitReason: String,
+    exitCode: Int)
+    extends TaskFailedReason {
+  override def toErrorString: String = s"Task with ID $taskId had its executor ${execId}" +
+    s" exit normally with exit code $exitCode, due to the following reason: $exitReason."
```

Comment:
This is a little verbose.

`exit normally with exit code $exitCode:\n $exitReason`